### ISSUE TITLE
[FEA] test the UCX shuffle with the new build changes

### ISSUE THREAD

**Issue by abellina** [NVIDIA/spark-rapids#3450]
https://github.com/NVIDIA/spark-rapids/issues/3223 introduces a whole new way to build/package our plugin. The RAPIDS Shuffle Manage needs to be tested so we make sure it works with all versions of  Spark with all the diffs performed.

We also want to add here that as of the writing of this issue, the RAPIDS Shuffle Manager is lazily initialized in the driver. A test in the shell shows that it gets initialized when it is needed to register some shuffle, and we output the following at the worst time:

```
2021-09-09 04:04:08,563 WARN  RapidsShuffleInternalManager:69 - Rapids Shuffle Plugin enabled. Transport enabled (remote fetches will use com.nvidia.spark.rapids.shuffle.ucx.UCXShuffleTransport. To disable the RAPIDS Shuffle Manager set `spark.rapids.shuffle.enabled` to false
```

I'll add a change to make this happen earlier during DriverPlugin initialization also

**Comment by abellina** [comment]
So far so good. I've tested with: 3.0.1, 3.0.2, 3.0.3, 3.0.4, 3.1.1, 3.1.2, 3.1.3, 3.2.0, and databricks 7.3 and 8.2. The last one that is remaining is CDH. 

I tested by running a shell and running a simple query, and verifying that the shuffle manager loads. One small issue I had seen before is handled by https://github.com/NVIDIA/spark-rapids/pull/3493, but otherwise this seems to be working fine.
